pkg/controller/storagecluster/reconcile.go

```diff
@@ -271,10 +231,11 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 				reqLogger.Error(phaseErr, "Failed to set PhaseClusterExpanding")
 			}
 		} else {
-			if instance.Status.Phase != statusutil.PhaseReady {
+			if instance.Status.Phase != statusutil.PhaseReady &&
+				instance.Status.Phase != statusutil.PhaseConnecting &&
+				instance.Status.Phase != statusutil.PhaseConnected {
 				instance.Status.Phase = statusutil.PhaseProgressing
-				phaseErr := r.client.Status().Update(context.TODO(), instance)
-				if phaseErr != nil {
+				if phaseErr := r.client.Status().Update(context.TODO(), instance); phaseErr != nil {
```

Comment:
I see what you're going for here, but I prefer the way it was before; it's a little easier to parse.

Reply:
My main intention was, as we are not using `phaseErr` later, initializing it within if statement make it local and the variable will be marked immediately for Garbage Collection , once the if scope is done...
Reverted back to the previous way... (=

Reply:
In general I'm in favor of that, honestly. But in this case, the call `r.client.Status().Update(context.TODO(), instance)` is a bit too long and complex in punctuation. If it were something like `r.updateStorageCluster()` that would be more reasonable to me. I'm not suggesting it as a change here, mind you, just giving examples of when I would and would not advocate for this kind of optimization.

Reply:
Also, maybe I'm missing something, but I still see the commit as being `if phaseErr := FOO; phaseErr != nil`, rather than the previous split.

Reply:
Sorry, missed the change... Added it...